### PR TITLE
Feat: cwm slider input

### LINKED ISSUE #1635: Create a slider input
# Description

With the new way of generating inputs made possible by #1619 we can improve our inputs. We would like to add a [slider input](https://mui.com/material-ui/react-slider/). 

It should at least have the following customization options:

- min
- max
- step
- unit

The config object for rendering the inputs can be found at [`src/models/themePlaygroundOptions.js`](https://github.com/Greenstand/treetracker-web-map-client/blob/61a9e2df258a9194b0c9df208e3c08686731bb1e/src/models/themePlaygroundOptions.js).
The place where the inputs are being rendered is [`src/components/playground/PropInput.js`](https://github.com/Greenstand/treetracker-web-map-client/blob/61a9e2df258a9194b0c9df208e3c08686731bb1e/src/components/playground/PropInput.js)

For building this I would recommend to first test the slider on one of the `fontSize` properties in the typography section.

NOTE: when working on this you should, checkout from the `cwm` branch, something like:

```sh
git checkout cwm
git checkout -b feat/cwm/slider-input
```

Feel free to ask any questions

The theme editor can be found on http://localhost:3000/admin/theme

---

Some hints:

- Please read our [readme](https://github.com/Greenstand/treetracker-web-map-client#treetracker-web) for more information/guide/tutorial.
- For design guidelines checkout our [Figma design](https://www.figma.com/file/XdYFdjlsHvxehlrkPVYq0l/Greenstand-Webmap).
- Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
- To know more about our organization, visit our [website](https://greenstand.org).
- If you want to join the slack community (some resources need the community member's permission), please leave your email address.

### PR DESCRIPTION
**Pull request by khalatevarun**
# Description

[comment]: # 'Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.'

* Adds a slider input component to replace textinput to improve user experience and reduce human errors, starting with fontSize

Fixes #1635 

## Type of change

[comment]: # 'Please delete options that are not relevant.'

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots

image

https://github.com/Greenstand/treetracker-web-map-client/assets/68511458/7302039c-32c0-40f2-81e3-0f02717fe74b

# How Has This Been Tested?

- [ ] Cypress integration
- [ ] Cypress component tests
- [ ] Jest unit tests

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by khalatevarun** [comment]
@RubenSmn I have started with the development.
For now I have added changes to h1, could you please run it once on local and check if the basic functionality is working?

Listing down few questions here
* How should be the UI? - Currently, I have just added a new Slider input above TextInput and both the values will be in sync with single state variable. TextInput will not be editable since its to convey the value on the slider.

* Should we name this additional configs ( max, min, unit, step ) as `configs`? 

Would really appreciate your input on this @RubenSmn

File: src/components/playground/PropInput.js [hunk 1]
```diff
@@ -22,6 +23,9 @@ export default function PropInput({ prop, path, propName }) {
         <FontWeightInput prop={prop} pathToProp={path} propName={propName} />
       );
       break;
+    case CONFIG_INPUT_TYPES.FONT_SIZE:
```
Comment (RubenSmn) [inline src/components/playground/PropInput.js:26]: No need to add a specific input type for the font size, in the config we can simply change the `inputType` to a `CONFIG_INPUT_TYPES.SLIDER`

**Review by RubenSmn — CHANGES_REQUESTED** [review]
Hi @khalatevarun great to see you working on this!

> How should be the UI? - Currently, I have just added a new Slider input above TextInput and both the values will be in sync with single state variable. TextInput will not be editable since its to convey the value on the slider.

I think the slider is just fine, it has a indicator of the current value. We might want to add some marks as a extra indication.

> Should we name this additional configs ( max, min, unit, step ) as configs?

Yes this is a good addition, it is easier to simply pass the props for the slider as `{...prop?.configs}`. But we also need to append the `unit` to the new value once it has changed. I would like to suggest something like this

```javascript
const { unit, ...restOfInputProps } = prop?.inputProps;

// ...

<Slider
  // ...
  {...restOfInputProps}
/>
```

I think a more descriptive name would be `inputProps` instead of `configs`.

**Comment by khalatevarun** [comment]
https://github.com/Greenstand/treetracker-web-map-client/assets/68511458/9b5769e0-5bca-402e-be56-dc48358ac09e

@RubenSmn please check the latest changes in the video here, did you mean something like this?
I have also updated the changes in the PR

**Review by RubenSmn — CHANGES_REQUESTED** [review]
This looks really good @khalatevarun :ok_hand: very nice. I found two things we can improve upon:

- the initial value is not correctly displayed on the slider, the initial value of the h1 is `48px` and the slider has a initial value of `20px`
- the reset button is missing

As a reference you can take a look at the `TextInput` component :wink: I made a custom hook which allows you to access the default value.

**Comment by khalatevarun** [comment]
Done, you can check now @RubenSmn

**Review by RubenSmn — COMMENTED** [review]
This is working very nice, I think we can update the config file and change the inputs to a slider input where suited!

**Comment by dadiorchen** [comment]
Good job, guys!

**Review by RubenSmn — APPROVED** [review]
Thanks, awesome work!

**Pull request merged**